proto/cometbft/abci/v4/types.proto

```diff
@@ -53,19 +53,23 @@ message Request {
   reserved 4, 7, 9, 10; // SetOption, BeginBlock, DeliverTx, EndBlock
 }
 
+// Type of the transaction check request.
+//
+// This enumeration is incompatible with the CheckTxType definition in
+// cometbft.abci.v1 and therefore shall not be used in encoding with the same
+// field number.
 enum CheckTxType {
   option (gogoproto.goproto_enum_prefix) = false;
 
-  // The non-conventional name for the zero value is preserved
-  // due to history of use.
-  // buf:lint:ignore ENUM_ZERO_VALUE_SUFFIX
-  CHECK_TX_TYPE_NEW = 0;
-  CHECK_TX_TYPE_RECHECK = 1;
+  CHECK_TX_TYPE_UNKNOWN = 0;
+  CHECK_TX_TYPE_NEW = 1;
+  CHECK_TX_TYPE_RECHECK = 2;
```

Comment:
Is it possible to:
* keep RECHECK with the same number as before (still breaking change, but one less element for devs to take care of when adapting their code)
* Rename NEW to something like CHECK, so reduce the possibility of confusion when referring to either (or both) in discussions

Reply:
> Would it be possible to:
> 
>     * keep RECHECK with the same number as before (still breaking change, but one less element for devs to take care of when adapting their code)

I don't think it will matter when adapting code, unless someone's been using literal numbers to represent these codes. The symbolic constants are renamed anyway. But the order of numbers does not matter, so why not.

>     * Rename NEW to something like CHECK, to reduce the possibility of confusion when referring to either (or both) in discussions

Good idea, I overlooked it from your proposal. Will do.

Reply:
Changed in cbf1e802b5b9b13ad68963dae3266d810c87c447